geode-core/src/main/java/org/apache/geode/cache/query/internal/CompiledComparison.java

```diff
@@ -651,6 +655,29 @@ public static String getSimpleClassName(Class cls) {
         indexData = QueryUtils.getAvailableIndexIfAny(path, context, this._operator);
       }
 
+      // Do not use indexes when map index with allkeys and != condition or when comparing with null
+      try {
+        if (indexData != null
```

Comment:
Would it be possible to extract the contents of this if condition to a method, and refactor it slightly to make it easier to parse and understand? As it currently is, it's very difficult to see what the desired condition is. Readability is more important than brevity here, I think.

Reply:
Yes. I totally agree.